backend/application/syson-application/src/test/java/org/eclipse/syson/application/controllers/diagrams/general/view/GVTopNodeCreationTests.java

```diff
@@ -207,4 +218,78 @@ public void createTopNode(EClass eClass, int compartmentCount) {
+                }
+        ));
+
+        String creationToolId = this.diagramDescriptionIdProvider.getDiagramCreationToolId(this.descriptionNameGenerator.getCreationToolName(SysmlPackage.eINSTANCE.getNamespaceImport()));
+        if (libId.get().isPresent()) {
+            this.verifier.then(() -> {
+                this.nodeCreationTester.createNode(SysMLv2Identifiers.GENERAL_VIEW_EMPTY_PROJECT,
+                        this.diagram,
+                        null,
+                        creationToolId,
+                       List.of(new ToolVariable("SelectedObject", libId.get().get(), ToolVariableType.OBJECT_ID)));
+            });
+
+            Consumer<DiagramRefreshedEventPayload> updatedDiagramConsumer = payload -> Optional.of(payload)
+                    .map(DiagramRefreshedEventPayload::diagram)
+                    .ifPresentOrElse(newDiagram -> {
+                        new CheckDiagramElementCount(this.diagramComparator)
+                                .hasNewEdgeCount(0)
+                                .hasNewNodeCount(1)
+                                .check(this.diagram.get(), newDiagram);
+
+                        String newNodeDescriptionName = this.descriptionNameGenerator.getNodeName(SysmlPackage.eINSTANCE.getNamespaceImport());
+
+                        new CheckNodeOnDiagram(this.diagramDescriptionIdProvider, this.diagramComparator)
+                                .hasNodeDescriptionName(newNodeDescriptionName)
+                                .hasCompartmentCount(0)
+                                .check(this.diagram.get(), newDiagram);
+                    }, () -> fail("Missing diagram"));
+
+            this.verifier.consumeNextWith(updatedDiagramConsumer);
+        }
+    }
+
+    private Optional<String> getISQAcousticsLibraryId(IEditingContext editingContext) {
+        Optional<ResourceSet> optionalResourceSet = Optional.of(editingContext)
+                .filter(IEMFEditingContext.class::isInstance)
+                .map(IEMFEditingContext.class::cast)
+                .map(IEMFEditingContext::getDomain)
+                .map(EditingDomain::getResourceSet);
+        if (optionalResourceSet.isPresent()) {
+            var resourceSet = optionalResourceSet.get();
+            var optionalPackage = resourceSet.getResources().stream()
+                    .filter(resource -> resource.eAdapters().stream()
+                            .filter(ResourceMetadataAdapter.class::isInstance)
+                            .map(ResourceMetadataAdapter.class::cast)
+                            .findFirst()
+                            .map(ResourceMetadataAdapter::getName).stream()
+                            .anyMatch("ISQAcoustics"::equals))
+                    .findFirst().map(resource -> {
+                        AtomicReference<Package> result = new AtomicReference<>();
+                        resource.getAllContents().forEachRemaining(eObject -> {
```

Comment:
You shouldn't have to do an `eAllContents` here. The `ISQAcoustics` should be at the root of the resource or just below.
Please rework the algorithm.

Reply:
Done